lib/internal/per_context/primordials.js

```diff
@@ -124,5 +140,16 @@ primordials.SafePromise = makeSafe(
   copyPrototype(original.prototype, primordials, `${name}Prototype`);
 });
 
+// Create copies of intrinsic objects that require a valid `this` to call
+// static methods.
+[
+  'Promise',
```

Comment:
It might be more informative to reference https://www.ecma-international.org/ecma-262/#sec-promise.all here which has this note:

> The all function requires its this value to be a constructor function that supports the parameter conventions of the Promise constructor.